### PR TITLE
ARROW-12058: [Python] Enable arithmetic operations on Expressions

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: python/pyarrow/tests/test_dataset.py [hunk 1]
```diff
@@ -472,6 +472,14 @@ def test_expression_boolean_operators():
         not true
 
 
+def test_expression_arithmetic_operators():
+    field = ds.field("field")
+    field + 1
+    field - 1
+    field * 1
+    field / 1
```
Comment (jorisvandenbossche) [inline python/pyarrow/tests/test_dataset.py:480]: This is a bit a dummy test until we can actually use those expressions (eg in https://github.com/apache/arrow/pull/9676)

Comment (lidavidm) [reply python/pyarrow/tests/test_dataset.py:480]: Now that #9676 is merged, do you want to update the test?

**Review by lidavidm — APPROVED** [review]

File: python/pyarrow/tests/test_dataset.py (hunk 1 above)
Comment (kszucs) [reply python/pyarrow/tests/test_dataset.py:480]: ping @jorisvandenbossche

Comment (jorisvandenbossche) [reply python/pyarrow/tests/test_dataset.py]: Updated the test now

**Review by lidavidm — APPROVED** [review]
LGTM

**Comment by lidavidm** [comment]
@jorisvandenbossche are you still planning on ARROW-12060 (compute functions on expressions) or should someone else look?

**Comment by jorisvandenbossche** [comment]
I had tabled that issue somewhat because I was assuming that @bkietz would do it soon at the C++ level anyway with ARROW-11929. But if that isn't planned to be worked on soon, it might still be worth adding some functionality in python/cython for it.

**Comment by lidavidm** [comment]
Thanks, makes sense (especially since without moving Expression, then you'd need some work to avoid a hard dependency from compute on dataset in Python).

**Pull request closed without merging**